File: Python/EstruturaDeDecisao/exercicio_28.py
"""
O Hipermercado Tabajara está com uma promoção de carnes que é imperdível. Confira:

                      Até 5 Kg           Acima de 5 Kg
File Duplo      R$ 4,90 por Kg          R$ 5,80 por Kg
Alcatra         R$ 5,90 por Kg          R$ 6,80 por Kg
Picanha         R$ 6,90 por Kg          R$ 7,80 por Kg

Para atender a todos os clientes, cada cliente poderá levar apenas um dos tipos de carne da promoção, porém não há limites para a quantidade de carne por cliente.
Se compra for feita no cartão Tabajara o cliente receberá ainda um desconto de 5% sobre o total da compra.
Escreva um programa que peça o tipo e a quantidade de carne comprada pelo usuário e gere um cupom fiscal.
Contendo as informações da compra: tipo e quantidade de carne, preço total, tipo de pagamento, valor do desconto e valor a pagar.
"""

tipos_carne = {
    1: 'File Duplo',
    2: 'Alcatra',
    3: 'Picanha'}
tabela_precos = {
    'File Duplo': {5: 4.90, 'acima': 5.80},
    'Alcatra': {5: 5.90, 'acima': 6.80},
    'Picanha': {5: 6.90, 'acima': 7.80}
}
print("Tipos de carne disponíveis:")
print("1 - File Duplo")
print("2 - Alcatra")
print("3 - Picanha")

tipo_carne_escolhida = int(input("Digite o número correspondente ao tipo de carne que deseja comprar: "))
tipo_carne = tipos_carne.get(tipo_carne_escolhida)
kg_carne = float(input("Digite a quantidade (em Kg) de carne que deseja comprar: "))
cartao_tabajara = input("Você possui o cartão Tabajara? (sim ou não): ").lower()

if kg_carne <= 5:
    preco_por_kg = tabela_precos[tipo_carne][5]
else:
    preco_por_kg = tabela_precos[tipo_carne]['acima']

valor = kg_carne * preco_por_kg

desconto = 0
if cartao_tabajara == 'sim':
    desconto = valor * 0.05
    valor -= desconto

print("\nCupom Fiscal")
print(f"Tipo de carne: {tipo_carne}")
print(f"Quantidade: {kg_carne:.1f}kg")
print(f"Preço total: R${valor:.2f}")
print(f"Tipo de pagamento: {'Cartão Tabajara' if cartao_tabajara == 'sim' else 'Dinheiro'}")
print(f"Desconto: R${desconto:.2f}")
print(f"Valor a pagar: R${valor:.2f}")

File: Java/EstruturaDeDecisao/Exercicio28.java
// O Hipermercado Tabajara está com uma promoção de carnes que é imperdível. Confira:

//                       Até 5 Kg           Acima de 5 Kg
// File Duplo      R$ 4,90 por Kg          R$ 5,80 por Kg
// Alcatra         R$ 5,90 por Kg          R$ 6,80 por Kg
// Picanha         R$ 6,90 por Kg          R$ 7,80 por Kg

// Para atender a todos os clientes, cada cliente poderá levar apenas um dos tipos de carne da promoção, porém não há limites para a quantidade de carne por cliente.
// Se compra for feita no cartão Tabajara o cliente receberá ainda um desconto de 5% sobre o total da compra.
// Escreva um programa que peça o tipo e a quantidade de carne comprada pelo usuário e gere um cupom fiscal,
// contendo as informações da compra: tipo e quantidade de carne, preço total, tipo de pagamento, valor do desconto e valor a pagar.
package EstruturaDeDecisao;
import java.util.Scanner;

public class Exercicio28 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        double descontoCartaoTabajara = 0.05;

        System.out.println("Tipos de carne disponíveis:");
        System.out.println("1 - File Duplo");
        System.out.println("2 - Alcatra");
        System.out.println("3 - Picanha");
        System.out.print("Digite o número correspondente ao tipo de carne que deseja comprar: ");
        int tipoCarne = scanner.nextInt();

        if (tipoCarne < 1 || tipoCarne > 3) {
            System.out.println("Opção inválida. Saindo do programa.");
            scanner.close();
            return;
        }

        System.out.print("Digite a quantidade (em Kg) de carne que deseja comprar: ");
        double quantidadeCarne = scanner.nextDouble();

        double precoPorKg = 0;
        String tipoCarneDescricao = "";

        switch (tipoCarne) {
            case 1:
                tipoCarneDescricao = "File Duplo";
                if (quantidadeCarne <= 5) {
                    precoPorKg = 4.90;
                } else {
                    precoPorKg = 5.80;
                }
                break;
            case 2:
                tipoCarneDescricao = "Alcatra";
                if (quantidadeCarne <= 5) {
                    precoPorKg = 5.90;
                } else {
                    precoPorKg = 6.80;
                }
                break;
            case 3:
                tipoCarneDescricao = "Picanha";
                if (quantidadeCarne <= 5) {
                    precoPorKg = 6.90;
                } else {
                    precoPorKg = 7.80;
                }
                break;
            default:
                System.out.println("Opção inválida. Saindo do programa.");
                scanner.close();
                return;
        }

        double valorTotal = quantidadeCarne * precoPorKg;

        System.out.print("Deseja pagar com o cartão Tabajara? (S/N): ");
        String cartaoTabajara = scanner.next().toUpperCase();

        double valorDesconto = 0;
        String formaPagamento = "";
        if (cartaoTabajara == "S") {
            valorDesconto = valorTotal * descontoCartaoTabajara;
            valorTotal -= valorDesconto;
            formaPagamento = "Cartão Tabajara";
        } else if (cartaoTabajara == "N") {
            formaPagamento = "Dinheiro";
        } else {
            System.out.println("Forma de pagamento inválida.");
            scanner.close();
            return;
        }

        System.out.println("Cupom Fiscal:");
        System.out.println("Tipo de carne: " + tipoCarneDescricao);
        System.out.println("Quantidade: " + quantidadeCarne + " Kg");
        System.out.printf("Preço total: R$ %.2f%n", valorTotal);
        System.out.println("Tipo de pagamento: " + formaPagamento);
        System.out.printf("Valor do desconto: R$ %.2f%n", valorDesconto);
        System.out.printf("Valor a pagar: R$ %.2f%n", valorTotal);

        scanner.close();
    }
}
